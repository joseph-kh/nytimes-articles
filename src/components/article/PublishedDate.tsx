import React from "react";
import { StyleSheet } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import Row from "@/components/ui/containers/Row";
import RegularText from "@/components/ui/texts/RegularText";
import { horizontalScale, moderateScale } from "@/utils/scaler";
import COLORS from "@/theme/colors";

type Props = {
  publishedDate: string;
};

const PublishedDate: React.FC<Props> = (props) => {
  const { publishedDate } = props;

  return (
    <Row>
      <MaterialCommunityIcons
        name="calendar-today"
        size={moderateScale(24)}
        style={styles.calendarStyle}
      />
      <RegularText style={styles.textStyle}>{publishedDate}</RegularText>
    </Row>
  );
};

export default PublishedDate;

const styles = StyleSheet.create({
  calendarStyle: {
    color: COLORS.gray,
    marginRight: horizontalScale(5),
  },

  textStyle: {
    color: COLORS.gray,
  },
});
