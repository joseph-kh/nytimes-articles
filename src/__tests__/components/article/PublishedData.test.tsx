import React from "react";
import renderer from "react-test-renderer";
import PublishedDate from "@/components/article/PublishedDate";

describe("PublishedDate Component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<PublishedDate publishedDate="2023-09-03" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
