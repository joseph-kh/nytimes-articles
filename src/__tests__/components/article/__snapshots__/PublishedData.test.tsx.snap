// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishedDate Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text />
  <Text
    style={
      {
        "color": "#7e7e7e",
        "fontFamily": "Regular",
        "fontSize": 21,
      }
    }
  >
    2023-09-03
  </Text>
</View>
`;
